hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferService.java

```diff
@@ -164,33 +165,30 @@ public RingbufferContainer getContainer(String name) {
             return ringbuffer;
         }
 
-        RingbufferConfig ringbufferConfig = getRingbufferConfig(name);
         ringbuffer = new RingbufferContainer(
                 name,
-                ringbufferConfig,
+                getRingbufferConfig(name),
                 nodeEngine.getSerializationService(),
                 nodeEngine.getConfigClassLoader());
         ringbuffer.getStore().instrument(nodeEngine);
         containers.put(name, ringbuffer);
         return ringbuffer;
     }
 
-    private RingbufferConfig getRingbufferConfig(String name) {
-        Config config = nodeEngine.getConfig();
-        return config.getRingbufferConfig(getConfigName(name));
-    }
-
     public void addRingbuffer(String name, RingbufferContainer ringbuffer) {
         checkNotNull(name, "name can't be null");
         checkNotNull(ringbuffer, "ringbuffer can't be null");
-        final RingbufferConfig config = nodeEngine.getConfig().getRingbufferConfig(name);
-        final SerializationService serializationService = nodeEngine.getSerializationService();
         ringbuffer.init(
                 name,
-                config,
-                serializationService,
+                getRingbufferConfig(name),
+                nodeEngine.getSerializationService(),
                 nodeEngine.getConfigClassLoader());
         ringbuffer.getStore().instrument(nodeEngine);
         containers.put(name, ringbuffer);
     }
+
+    private RingbufferConfig getRingbufferConfig(String name) {
+        Config config = nodeEngine.getConfig();
+        return config.findRingbufferConfig(getConfigName(name));
```

Comment:
Looks fine to me. The only thing I've noticed now, not so much related to this change, is that there is a slight difference in using `find*Config` and `get*Config`. If there is no config that matches by pattern, `get*Config` will add a new config with the requested name but based on the default config (and add the default config if necessary). On the other hand `find*Config` will not add this new config with the requested name. Usually there's no repercussion but I need to think about it in the context of dynamic configs.